packages/dart/lib/src/span_v2/noop_span.dart

```diff
@@ -0,0 +1,25 @@
+import '../../sentry.dart';
+import 'span.dart';
+import 'span_v2_status.dart';
+
+class NoOpSpan implements Span {
+  const NoOpSpan();
+
+  @override
+  void end({DateTime? endTimestamp}) {}
+
+  @override
+  void setAttribute(String key, SentryAttribute value) {}
```

Comment:
We can also set `undefined` per API: `setAttribute(key: string, value: SpanAttributeValue | undefined): this;`. Should make this optional, so users can remove keys.